src/App.tsx

```diff
@@ -16,35 +26,121 @@ export const goodsFromServer = [
 ];
 
 export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([...goodsFromServer]);
+  const [sortState, setSortState] = useState<{
+    reversed: boolean;
+    criterion: SortType;
+  }>({
+    reversed: false,
+    criterion: SortType.None,
+  });
+  const [isReset, setIsReset] = useState(false as boolean);
+
+  const sortGoodsByAlphabet: ClickButton = () => {
+    const sortedGoods: Good[] = [...goods].sort((a, b) => {
+      if (sortState.reversed) {
+        return b.localeCompare(a);
+      }
+
+      return a.localeCompare(b);
+    });
+
+    setGoods(sortedGoods);
+    setSortState({ ...sortState, criterion: SortType.Alphabet });
+    setIsReset(true);
+  };
+
+  const sortGoodsByLength: ClickButton = () => {
+    const sortedGoods = [...goods].sort((a, b) => {
+      if (sortState.reversed) {
+        return b.length - a.length;
+      }
+
+      return a.length - b.length;
+    });
+
+    setGoods(sortedGoods);
+    setSortState({ ...sortState, criterion: SortType.Length });
+    setIsReset(true);
+  };
+
+  const reverseGoods: ClickButton = () => {
+    setSortState(prevState => {
+      const reversed = !prevState.reversed;
+
+      setGoods(prevGoods => {
+        const newGoods = [...prevGoods].reverse();
+
+        setIsReset(
+          !newGoods.every(
+            (element, index) => element === goodsFromServer[index],
+          ),
+        );
+
+        return newGoods;
+      });
+
+      return { ...prevState, reversed };
+    });
+  };
+
+  const resetGoods: ClickButton = () => {
+    setGoods([...goodsFromServer]);
+    setSortState({
+      reversed: false,
+      criterion: SortType.None,
+    });
+    setIsReset(false);
```

Comment:
The `resetGoods` function correctly resets the `goods` and `sortState`, but ensure that this is the intended behavior for `isReset`. If `isReset` is meant to indicate whether the list is in its original state, the logic should be consistent across the component.